src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/TransactionTest/TransactionInferenceTest.cs

```diff
@@ -0,0 +1,372 @@
+
+                        Helpers.InsertTable(connection, table); // insert 1 row
+
+                        Assert.Equal(1, Helpers.CountTable(connection, table)); // verify row count = 1
+
+                        transaction.Rollback(); // rollback cleanly
+
+                        Assert.Equal(0, Helpers.CountTable(connection, table)); // verify row count = 0
+                    }
+
+                    Helpers.PurgeTable(connection, table);
+
+                    Assert.Throws<InvalidProgramException>(new Action(() =>
+                    {
+                        using (SqlTransaction transaction = connection.BeginTransaction())
+                        {
+                            // run several validations against command/connection/transaction
+                            Assert.Equal(connection, transaction.Connection);
+                            Assert.Equal(connection, command.Connection);
+                            Assert.Equal(transaction, command.Transaction);
+
+                            Helpers.InsertTable(connection, table); // insert 1 row
+
+                            Assert.Equal(1, Helpers.CountTable(connection, table)); // verify row count = 1
+
+                            throw new InvalidProgramException(); // rollback via exception
+                        }
+                    }));
+                    Assert.Equal(0, Helpers.CountTable(connection, table)); // verify row count = 0
+
+                    using (SqlTransaction transaction = connection.BeginTransaction())
+                    {
+                        // run several validations against command/connection/transaction
+                        Assert.Equal(connection, transaction.Connection);
+                        Assert.Equal(connection, command.Connection);
+                        Assert.Equal(transaction, command.Transaction);
+
+                        Helpers.InsertTable(connection, table); // insert 1 row
+
+                        Assert.Equal(1, Helpers.CountTable(connection, table)); // verify row count = 1
+
+                        transaction.Commit(); // commit
+
+                        Assert.Equal(1, Helpers.CountTable(connection, table)); // verify row count = 0
+                    }
+                }
+            }
+        }
+        #endregion
+
+        #region BulkCopy
+        // To be continued.
+        #endregion
+
+        #region Helpers
+        /// <summary>
+        /// Helper methods used in the Test methods to reduce code bloat.
+        /// </summary>
+        internal static class Helpers
+        {
+            public static string ConnectionString => CreateConnectionString();
+
+            public static string CreateConnectionString(Action<SqlConnectionStringBuilder> builder = null)
+            {
+                var csb = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString)
+                {
+                    MultipleActiveResultSets = false,
```

Comment:
Your change has affected the MARS behavior. Please, add more test units to cover it.